protocol/v2/ssv/validator/validator.go

```diff
@@ -77,21 +75,36 @@ func NewValidator(pctx context.Context, cancel func(), options Options) *Validat
 }
 
 // StartDuty starts a duty for the validator
-func (v *Validator) StartDuty(duty *spectypes.Duty) error {
+func (v *Validator) StartDuty(logger *zap.Logger, duty *spectypes.Duty) error {
+	logger = logger.Named("StartDuty")
+
 	dutyRunner := v.DutyRunners[duty.Type]
 	if dutyRunner == nil {
 		return errors.Errorf("duty type %s not supported", duty.Type.String())
 	}
-	return dutyRunner.StartNewDuty(duty)
+	return dutyRunner.StartNewDuty(logger, duty)
 }
 
 // ProcessMessage processes Network Message of all types
-func (v *Validator) ProcessMessage(msg *queue.DecodedSSVMessage) error {
+func (v *Validator) ProcessMessage(logger *zap.Logger, msg *queue.DecodedSSVMessage) error {
+	logger = logger.Named("ProcessMessage")
+
 	dutyRunner := v.DutyRunners.DutyRunnerForMsgID(msg.GetID())
 	if dutyRunner == nil {
 		return errors.Errorf("could not get duty runner for msg ID")
 	}
 
+	if br := dutyRunner.GetBaseRunner(); br != nil && br.State != nil {
+		var duty *spectypes.Duty
+		if dv := br.State.DecidedValue; dv != nil && dv.Duty != nil {
+			duty = dv.Duty
+		} else if d := br.State.StartingDuty; d != nil {
+			duty = d
+		}
+
+		logger = logger.With(zap.String("taskUniqueID", getTaskUniqueID(dutyRunner, duty)))
```

Comment:
why do we only set the logger only in this `if` condition?

Reply:
@aaron-blox To avoid nil pointer dereference in tests

Reply:
what is nil? duty? do we process messages with a nil duty? if we do, we should still set a trace ID for them

Reply:
an example is `preconsensus.NoRunningDuty` in spec tests (state is nil), we don't start a duty, process a message and its validation fails, but we set a trace ID before the validation